cpp/include/cudf/column/column.hpp

```diff
@@ -75,6 +76,30 @@ class column {
    */
   column(column&& other) noexcept;
 
+  /**
+   * @brief Construct a new column by taking ownership of the contents of a device_uvector.
+   *
+   * @param other The device_uvector whose contents will be moved into the new column.
+   * @param null_mask Optional, column's null value indicator bitmask. May
+   * be empty if `null_count` is 0 or `UNKNOWN_NULL_COUNT`.
+   * @param null_count Optional, the count of null elements. If unknown, specify
+   * `UNKNOWN_NULL_COUNT` to indicate that the null count should be computed on
+   * the first invocation of `null_count()`.
+   */
+  template <typename T, CUDF_ENABLE_IF(cudf::is_numeric<T>() or cudf::is_chrono<T>())>
+  column(rmm::device_uvector<T>&& other,
+         rmm::device_buffer&& null_mask = {},
+         size_type null_count           = cudf::UNKNOWN_NULL_COUNT)
+    : _type{cudf::type_to_id<T>()},
+      _size{static_cast<size_type>(other.size())},
+      _data{other.release()},
+      _null_mask{std::forward<rmm::device_buffer>(null_mask)},
+      _null_count{null_count}
+  {
+    CUDF_EXPECTS(_size < std::numeric_limits<size_type>::max(),
+                 "Input vector exceeds maximum column capacity");
```

Comment:
In addition to the member init order I complained above, this check is also awkward. If the input vector has size exceeds the `size_type` limit, `_size` here will be overflow.

Reply:
It seems that a safe way to permanently avoid order issue (issue I mentioned above, order of calling `other.size()` and `other.release()`) and safely check `other.size()` within `size_type` limit is to initialize member variables inside the constructor block. This may be ugly but it is safe. Unless you guys have a better way to ensure safety.

Reply:
That's a good argument about checking `CUDF_EXPECTS(other.size() < std::numeric_limits<size_type>::max())`. The current `CUDF_EXPECTS` check doesn't really do anything because the `size_type _size` has already been initialized. We have to check against the `std::size_t` returned by `other.size()`. I'm fine with adapting this to initialize in the constructor block.